src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class FigureSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(5);
+        String color = colorSupplier.getRandomColor();
+
+        return switch (figureType) {
+            case 0 -> new Square(color, random.nextDouble() * 10);
+            case 1 -> new Rectangle(color, random.nextDouble() * 10, random.nextDouble() * 10);
+            case 2 -> new RightTriangle(color, random.nextDouble() * 10, random.nextDouble() * 10);
+            case 3 -> new Circle(color, random.nextDouble() * 10);
+            case 4 -> new IsoscelesTrapezoid(color, random.nextDouble() * 10,
+                    random.nextDouble() * 10, random.nextDouble() * 10);
```

Comment:
Instead of passing expressions directly in the constructor, create separate variables for the values and pass them on for better code readability.